### PR TITLE
EASI-1465/craco-uswds-upgrade

### PR DESCRIPTION
**Pull request by patrickseguraoddball**
# EASI-1465

## Changes and Description

- Upgraded to USWDS 3.3.0
- Upgraded to CRACO 7.0
- Upgraded react-scripts (CRA)
- Upgraded a few other packages for compatibility
- Removed craco-sass-resource-loader
- Removed old uswds theme package
- Adjust craco.config.js to compile new uswds 3.0
- Updated style entry point (index.scss) to import only used packages for better compile
- Imported uswds core in all stylesheet that utilized themes, functions, mixins, etc.
- Deprecated uswdsUtilities.scss
- Replaced EASI' `UsGovBanner` with Truss' `GovBanner`

No snapshots needed to be updated!

## Outstanding work:
Update Truss package (Will be in another ticket branch at this point)

I was decided to PoC on MINT for familiarity, but this upgrade can easily be adapted on EASI.

## How to test this change

- Make sure all tests run successfully, and try to spot any stylistic differences.  (Should be none)

## PR Author Review Checklist

- [ ] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

**Comment by DylanSpOddball** [comment]
It looks from the new `craco.config.js` that you're just using it to specify `includePaths`. The [CRA docs](https://create-react-app.dev/docs/adding-a-sass-stylesheet/) mention that you can set a `SASS_PATH` environment variable as a path to import modules from (i.e. `./src/stylesheets`), and use a `~` prefix to implement from `node_modules`. If you set `SASS_PATH` in the `.envrc` file and used `~` for `@use` statements pulling from `node_modules` (i.e. `@use '~uswds-core;'`), would that allow getting rid of Craco? Admittedly, I don't know if you still need it for other assets, like you mentioned working on in today (Thursday's) standup.

**Comment by patrickseguraoddball** [comment]
> It looks from the new `craco.config.js` that you're just using it to specify `includePaths`. The [CRA docs](https://create-react-app.dev/docs/adding-a-sass-stylesheet/) mention that you can set a `SASS_PATH` environment variable as a path to import modules from (i.e. `./src/stylesheets`), and use a `~` prefix to implement from `node_modules`. If you set `SASS_PATH` in the `.envrc` file and used `~` for `@use` statements pulling from `node_modules` (i.e. `@use '~uswds-core;'`), would that allow getting rid of Craco? Admittedly, I don't know if you still need it for other assets, like you mentioned working on in today (Thursday's) standup.

I gave it a shot but was running into the same issues outlined in this thread with using `SASS_PATH`

https://github.com/facebook/create-react-app/issues/12329

Most of the people ended up scrapping it.

**Comment by patrickseguraoddball** [comment]
Tests that require the three imported icons in `src/components/UsGovBanner/index.tsx` are failing.  I suspect it's because the USWDS alias's the path where the icons are stored after compilation.  Eslint does not recognize this alias, and there seems to be a discrepancy between eslint and the webpack compiler.

Importing from `@uswds/uswds/src/img/icon-dot-gov.svg` fulfills the compiler (recognizes the alias) but fails the linter
Importing from `@uswds/uswds/dist/img/icon-dot-gov.svg` fails the compiler (alias not needed) and passes the linter

File: src/components/Header/index.tsx [hunk 1]
```diff
@@ -2,13 +2,13 @@ import React, { useContext, useEffect, useRef, useState } from 'react';
...
 import { useOktaAuth } from '@okta/okta-react';
-import { IconMenu } from '@trussworks/react-uswds';
+import { GovBanner, IconMenu } from '@trussworks/react-uswds';
 import classnames from 'classnames';
 
 import { NavContext } from 'components/Header/navContext';
 import UswdsReactLink from 'components/LinkWrapper';
 import NavigationBar from 'components/NavigationBar';
-import UsGovBanner from 'components/UsGovBanner';
+// import UsGovBanner from 'components/UsGovBanner';
```
Comment (ClayBenson94) [inline src/components/Header/index.tsx:11]: Can we remove this commented code now that the source component is gone?

File: src/hooks/useCheckMobile.tsx [hunk 2]
```diff
@@ -1,6 +1,11 @@
 import { useEffect, useState } from 'react';
 
-import deviceBreakpoints from 'stylesheets/_variables.module.scss';
+// import deviceBreakpoints from '../stylesheets/_viewports.scss';
+const deviceBreakpoints = {
+  mobile: '480px',
+  tablet: '640px',
+  desktop: '1024px'
+};
```
Comment (ClayBenson94) [inline src/hooks/useCheckMobile.tsx:8]: Did something change with being able to import these viewports here? Looks like all the values are right in the viewports.scss file

**Review by ClayBenson94 — CHANGES_REQUESTED** [review]

**Review by ClayBenson94 — APPROVED** [review]
Changes look good!

**Pull request merged**